test/jdk/java/awt/print/PrinterJob/DrawStringMethods.java

```diff
@@ -21,44 +21,61 @@
  * questions.
  */
 
-/**
+import java.awt.Color;
+import java.awt.Font;
+import java.awt.Graphics;
+import java.awt.Graphics2D;
+import java.awt.font.FontRenderContext;
+import java.awt.print.Book;
+import java.awt.print.PageFormat;
+import java.awt.print.Printable;
+import java.awt.print.PrinterJob;
+import java.text.AttributedCharacterIterator;
+import java.text.AttributedString;
+
+import jtreg.SkippedException;
+
+/*
  * @test
  * @bug 4185019
  * @key printer
  * @summary Confirm that all of the drawString methods on Graphics2D
  *          work for printer graphics objects.
+ * @library /java/awt/regtesthelpers
+ * @library /test/lib
+ * @build PassFailJFrame
+ * @build jtreg.SkippedException
  * @run main/manual DrawStringMethods
  */
+public class DrawStringMethods implements Printable {
 
-import java.awt.*;
-import java.text.*;
-import java.awt.font.*;
-import java.awt.print.*;
+    private static final String INSTRUCTIONS =
+            " Confirm that the methods are printed.\n" +
+            " For Graphics: drawString, drawString, drawChars, drawBytes\n" +
+            " For Graphics2D: drawString, drawString, drawGlyphVector";
 
-public class DrawStringMethods implements Printable {
+    public static void main(String args[]) throws Exception {
 
-    public static void main(String args[]) {
-        String[] instructions =
-        {
-            "Confirm that the methods are printed.",
-            " For Graphics: drawString, drawString, drawChars, drawBytes",
-            " For Graphics2D: drawString, drawString, drawGlyphVector"
-        };
-        Sysout.createDialogWithInstructions( instructions );
+        if (PrinterJob.lookupPrintServices().length == 0) {
+            throw new SkippedException("Printer not configured or available."
+                    + " Test cannot continue.");
+        }
 
+        PassFailJFrame passFailJFrame = new PassFailJFrame.Builder()
+                .instructions(INSTRUCTIONS)
+                .rows((int) INSTRUCTIONS.lines().count() + 1)
+                .columns(45)
+                .build();
 
         PrinterJob pjob = PrinterJob.getPrinterJob();
```

Comment:
i think this could be moved into a helper function and called by passFailJFrame builder via .testui(DrawStringMethods::helperFunction)

Reply:
These class do not have any UI/frame, so I don't think this will workout.